icu4c/source/i18n/dtptngen.cpp

```diff
@@ -799,7 +799,11 @@ DateTimePatternGenerator::getCalendarTypeToUse(const Locale& locale, CharString&
             nullptr,
             FALSE,
             &err);
-        if (U_FAILURE(err)) { return; }
```

Comment:
A pattern that we use a lot in situations like this (where we expect a possible error) is to make a local error code.  Something like this:

```cpp
ErrorCode localStatus;
// You can now call both functions in sequence
ures_getFunctionalEquivalent(..., localStatus);
uloc_getKeywordValue(..., localStatus);
if (localStatus.isFailure() && localStatus != U_MISSING_RESOURCE_ERROR) {
  err = localStatus;
  return;
}
// save result to destination
```

The advantage is that you don't touch the "real" user-provided error code variable unnecessarily.

Search ICU source for "localStatus" for more examples.

Reply:
Thanks Shane!
I think that approach might be slightly cleaner as it would allow removing the check for failure twice. Though I think I'll use UErrorCode, as I don't want to pull in another header file, since I'm trying to keep this fix as small as possible (for others that will be patching things).
I'll push another commit with the change.

Reply:
@sffc I pushed another commit based on your suggestion. PTAL.
(I'll squash before merging if you think it looks okay).